server/contracts.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+
+	emulator "github.com/onflow/flow-emulator"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/templates"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-nft/lib/go/contracts"
+	fusd "github.com/onflow/fusd/lib/go/contracts"
+)
+
+var baseContractsPath = "./server/contracts/"
+
+type DeployDescription struct {
+	name        string
+	address     flow.Address
+	description string
+}
+
+func deployContracts(conf *Config, b *emulator.Blockchain) []DeployDescription {
+	ftAddress := flow.HexToAddress(fvm.FungibleTokenAddress(b.GetChain()).Hex())
+
+	fusdAddress, _ := deployContract("FUSD", fusd.FUSD(ftAddress.String()), b)
+	nftAddress, _ := deployContract("NonFungibleToken", contracts.NonFungibleToken(), b)
+	exampleNFT, _ := deployContract("ExampleNFT", contracts.ExampleNFT(nftAddress.Hex()), b)
+
+	nftStorefrontContract := loadContract("NFTStorefront.cdc", map[string]flow.Address{
+		"FungibleToken":    ftAddress,
+		"NonFungibleToken": nftAddress,
+	})
+
+	nftStorefront, _ := deployContract("NFTStorefront", nftStorefrontContract, b)
+
+	addresses := []DeployDescription{
+		{"FUSD", fusdAddress, "💵  FUSD contract"},
+		{"NonFungibleToken", nftAddress, "✨   NFT contract"},
+		{"ExampleNFT", exampleNFT, "✨   NFT contract"},
+		{"NFTStorefront", nftStorefront, "✨   NFT contract"},
+	}
+
+	return addresses
+}
+
+func loadContract(name string, replacements map[string]flow.Address) []byte {
+	code := string(readFile(filepath.Join(baseContractsPath, name)))
+	for name, realAddress := range replacements {
+		placeholder := regexp.MustCompile(fmt.Sprintf(`"[^"\s].*/%s.cdc"`, name))
+		code = placeholder.ReplaceAllString(code, fmt.Sprintf("0x%s", realAddress.String()))
+	}
+	return []byte(code)
+}
+
+func deployContract(name string, contract []byte, b *emulator.Blockchain) (flow.Address, error) {
```

Comment:
ah, interesting!! How do you deploy all at once?

Haha I thought it was all right to have different accounts for each one, but they could also go ito the same account... the bigger thing will probably be.. how do you control the accounts once you've deployed the contracts?
- actually it might be a good idea to make them all deploy from the service acct, since that's what you have the keys for, right?

Reply:
Yes, I think that it would be better to deploy on the service account not just because of the keys but also there are for sure a lot of tests/projects, unfortunately, depending on the sequence of addresses used for testing so if you will "use" those addresses to do these deployments we might break something for them. You can deploy them using a single transaction as contracts can be passed as an array, see here: https://pkg.go.dev/github.com/onflow/flow-go-sdk/templates#CreateAccount

Reply:
done!

![image](https://user-images.githubusercontent.com/30219253/139904785-88e76db8-d066-4a2b-9dc5-c7435abb99af.png)